tests/client/coll.rs

```diff
@@ -410,7 +447,11 @@ fn delete_many() {
     // Delete document
     coll.delete_many(doc2.clone(), None).ok().expect("Failed to delete documents.");
     let mut cursor = coll.find(None, None).ok().expect("Failed to execute find command.");
-    let result = cursor.next().unwrap();
+    let result = match cursor.next() {
+        Some(Ok(res)) => res,
+        Some(Err(_)) => panic!("Received error from 'cursor.next()'."),
+        None => panic!("Expected bson."),
```

Comment:
Err occurs when the cursor fails to retrieve more results from the server (via network and encoding/decoding error). Some(Ok()) and None() are the standard iterator pattern matches for more docs or none left. 

The iterator trait has an `Option<T>` return for `next()`, which is why the pattern is a little funky, as the Result has to be nested in.